packages/web3-eth-accounts/src/index.js

```diff
@@ -168,36 +154,28 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
             transaction.value = transaction.value || '0x';
             transaction.chainId = utils.numberToHex(transaction.chainId);
 
-            var rlpEncoded = RLP.encode([
-                Bytes.fromNat(transaction.nonce),
-                Bytes.fromNat(transaction.gasPrice),
-                Bytes.fromNat(transaction.gas),
-                transaction.to.toLowerCase(),
-                Bytes.fromNat(transaction.value),
-                transaction.data,
-                Bytes.fromNat(transaction.chainId || "0x1"),
-                "0x",
-                "0x"]);
-
-
-            var hash = Hash.keccak256(rlpEncoded);
+            if (privateKey.startsWith('0x')) {
+                privateKey = privateKey.substring(2);
+            }
 
-            var signature = Account.makeSigner(Nat.toNumber(transaction.chainId || "0x1") * 2 + 35)(Hash.keccak256(rlpEncoded), privateKey);
+            var ethTx = new Transaction(transaction);
+            ethTx.sign(Buffer.from(privateKey, 'hex'));
 
-            var rawTx = RLP.decode(rlpEncoded).slice(0, 6).concat(Account.decodeSignature(signature));
+            var validationResult = ethTx.validate(true);
 
-            rawTx[6] = makeEven(trimLeadingZero(rawTx[6]));
-            rawTx[7] = makeEven(trimLeadingZero(rawTx[7]));
-            rawTx[8] = makeEven(trimLeadingZero(rawTx[8]));
+            if (validationResult !== '') {
+                throw new Error('Signer Error: ' + validationResult);
+            }
 
-            var rawTransaction = RLP.encode(rawTx);
+            var rlpEncoded = ethTx.serialize().toString('hex');
+            var rawTransaction = '0x' + rlpEncoded;
+            var transactionHash = utils.keccak256(rawTransaction);
```

Comment:
Is this transactionHash line extra?

Reply:
I've added the ``transactionHash`` property to the returned object.